File: leetcode/medium/Q102_BinaryTreeLevelOrderTraversal.py
from typing import List

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

class Solution:
    def levelOrder(self, root: TreeNode) -> List[List[int]]:
        answer = []
        if not root:
            return answer
        que = [root]

        ans = []
        new_que = []
        while que:
            node = que.pop(0)
            ans.append(node.val)
            if node.left:
                new_que.append(node.left)
            if node.right:
                new_que.append(node.right)
            if not que:
                que = new_que.copy()
                new_que = []
                answer.append(ans.copy())
                ans = []
        return answer

File: src/medium/Q102_BinaryTreeLevelOrderTraversal.java
package medium;

import java.util.*;

public class Q102_BinaryTreeLevelOrderTraversal {

    public class TreeNode {
        int val;
        TreeNode left;
        TreeNode right;
        TreeNode() {}
        TreeNode(int val) { this.val = val; }
        TreeNode(int val, TreeNode left, TreeNode right) {
          this.val = val;
          this.left = left;
          this.right = right;
        }
    }

    class Solution {
        List<List<Integer>> answer;
        public List<List<Integer>> levelOrder(TreeNode root) {
            answer = new ArrayList<>();
            if(root != null){
                Queue<TreeNode> que = new LinkedList<>();
                que.add(root);
                while (!que.isEmpty()){
                    que = find(que);
                }
            }
            return answer;
        }

        public Queue<TreeNode> find(Queue<TreeNode> que){
            Queue<TreeNode> childQue = new LinkedList<>();
            List<Integer> valList = new ArrayList<>();
            while(!que.isEmpty()){
                TreeNode node = que.poll();
                valList.add(node.val);
                if(node.left != null) childQue.add(node.left);
                if(node.right != null) childQue.add(node.right);
            }
            answer.add(valList);
            return childQue;
        }
    }
}
